src/overlays/actors/ovl_En_Mm2/z_en_mm2.c

```diff
@@ -31,16 +30,57 @@ const ActorInit En_Mm2_InitVars = {
     (ActorFunc)EnMm2_Draw,
 };
 
-#endif
+#include "overlays/ovl_En_Mm2/ovl_En_Mm2.c"
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/EnMm2_Init.s")
+void EnMm2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnMm2* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/EnMm2_Destroy.s")
+    Actor_SetScale(&this->actor, 0.015f);
+    this->actionFunc = EnMm2_WaitForRead;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/func_809A2080.s")
+void EnMm2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/func_809A20FC.s")
+// Action function whilst Link is reading the letter.
+void EnMm2_Reading(EnMm2* this, GlobalContext* globalCtx) {
+    u8 temp_v0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/EnMm2_Update.s")
+    temp_v0 = func_80152498(&globalCtx->msgCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/EnMm2_Draw.s")
+    if (temp_v0 != 2) {
+        if (temp_v0 == 5 && func_80147624(globalCtx)) {
+            func_801477B4(globalCtx);
+            this->actionFunc = EnMm2_WaitForRead;
+        }
+    } else {
+        this->actionFunc = EnMm2_WaitForRead;
+    }
+}
+
+// Action function that awaits Link to read the letter, changing the A button to "Check" when he is within range to do
+// so (and facing the letter).
+void EnMm2_WaitForRead(EnMm2* this, GlobalContext* globalCtx) {
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        func_801518B0(globalCtx, 0x277BU, &this->actor);
+        this->actionFunc = EnMm2_Reading;
+    } else if ((this->actor.xzDistToPlayer < 60.0f) && (Actor_IsLinkFacingActor(&this->actor, 0x3000, globalCtx))) {
+        func_800B8614(&this->actor, globalCtx, 110.0f);
+    }
+}
+
+void EnMm2_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnMm2* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+}
+
+void EnMm2_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    GraphicsContext* gfxCtx;
+    EnMm2* this = THIS;
+
+    gfxCtx = globalCtx->state.gfxCtx;
+    func_8012C28C(globalCtx->state.gfxCtx);
+    gSPMatrix(gfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx), (0x00 | 0x02) | 0x00);
+    gSPDisplayList(gfxCtx->polyOpa.p++, sEnMm2DL);
+}
```

Comment:
In draw functions like this you usually need to use the `OPEN_DISPS`/`CLOSE_DISPS` macros and the `POLY_OPA_DISP` macro instead of accessing the `gfxCtx->polyOpa.p` element.
For more info on this you can check the tutorial (which is in a PR right now)
https://github.com/zeldaret/mm/pull/362/files
https://github.com/EllipticEllipsis/mm/blob/tutorial/docs/tutorial/draw_functions.md